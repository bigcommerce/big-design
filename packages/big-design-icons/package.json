{
  "name": "@bigcommerce/big-design-icons",
  "version": "1.2.2",
  "sideEffects": false,
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "typings": "dist/index.d.ts",
  "license": "(MIT AND CC-BY-3.0)",
  "author": "BigCommerce Inc",
  "repository": {
    "type": "git",
    "url": "https://github.com/bigcommerce/big-design.git",
    "directory": "packages/big-design-icons"
  },
  "scripts": {
    "build": "pnpm run build:cjs && pnpm run build:es && pnpm run build:dt",
    "build:cjs": "NODE_ENV=production BABEL_ENV=cjs babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/cjs",
    "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
    "build:dt": "tsc --emitDeclarationOnly",
    "build:icons": "node scripts/build.js",
    "build:flags": "node scripts/build-flags.js",
    "download": "node scripts/downloader.js",
    "lint": "eslint . --max-warnings 0",
    "typecheck": "tsc --noEmit"
  },
  "files": [
    "dist",
    "flags",
    "svgs"
  ],
  "publishConfig": {
    "access": "public"
  },
  "prettier": "@bigcommerce/eslint-config/prettier",
  "dependencies": {
    "@babel/runtime": "^7.26.0",
    "@bigcommerce/big-design-theme": "workspace:^"
  },
  "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "styled-components": "^5.3.5"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.7",
    "@babel/core": "^7.26.0",
    "@babel/plugin-transform-runtime": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@babel/preset-typescript": "^7.26.0",
    "@bigcommerce/configs": "workspace:^",
    "@styled/typescript-styled-plugin": "^1.0.1",
    "@svgr/core": "^8.1.0",
    "@svgr/plugin-jsx": "^8.1.0",
    "@svgr/plugin-prettier": "^8.1.0",
    "@svgr/plugin-svgo": "^8.1.0",
    "@types/react": "^18.2.73",
    "@types/react-dom": "^18.2.23",
    "@types/styled-components": "^5.1.34",
    "babel-plugin-styled-components": "^2.0.7",
    "camelcase": "^6.3.0",
    "flag-icons": "^7.3.2",
    "fs-extra": "^11.1.0",
    "glob": "^11.0.1",
    "inquirer": "^8.2.6",
    "inquirer-autocomplete-prompt": "^2.0.1",
    "prettier": "^2.4.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-is": "^18.2.0",
    "rimraf": "^6.0.1",
    "styled-components": "^5.3.11",
    "tiny-async-pool": "^2.0.1",
    "typescript": "^5.6.3"
  }
}
