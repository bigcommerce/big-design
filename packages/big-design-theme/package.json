{
  "name": "@bigcommerce/big-design-theme",
  "version": "1.1.0",
  "sideEffects": false,
  "main": "src/index.ts",
  "license": "(MIT AND CC-BY-3.0)",
  "author": "BigCommerce Inc",
  "repository": {
    "type": "git",
    "url": "https://github.com/bigcommerce/big-design.git",
    "directory": "packages/big-design-theme"
  },
  "scripts": {
    "build": "pnpm run build:cjs && pnpm run build:es && pnpm run build:dt",
    "build:cjs": "NODE_ENV=production BABEL_ENV=cjs babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/cjs",
    "build:es": "NODE_ENV=production BABEL_ENV=es babel --extensions \".ts,.tsx\" ./src --out-dir ./dist/es",
    "build:dt": "tsc -p tsconfig.declarations.json --emitDeclarationOnly",
    "lint": "eslint . --max-warnings 0",
    "prepack": "big-design-prepack",
    "postpack": "big-design-postpack",
    "test": "jest",
    "test:watch": "pnpm run test --watch",
    "typecheck": "tsc --noEmit"
  },
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public",
    "main": "dist/cjs/index.js",
    "module": "dist/es/index.js",
    "typings": "dist/index.d.ts"
  },
  "prettier": "@bigcommerce/eslint-config/prettier",
  "dependencies": {
    "@babel/runtime": "^7.26.0",
    "polished": "^4.3.1"
  },
  "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "styled-components": "^5.3.5"
  },
  "devDependencies": {
    "@babel/cli": "^7.25.7",
    "@babel/core": "^7.26.0",
    "@babel/plugin-transform-runtime": "^7.25.9",
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-react": "^7.26.3",
    "@babel/preset-typescript": "^7.26.0",
    "@bigcommerce/configs": "workspace:^",
    "@bigcommerce/pack": "workspace:^",
    "@styled/typescript-styled-plugin": "^1.0.1",
    "@swc/core": "^1.10.4",
    "@swc/jest": "^0.2.37",
    "@swc/plugin-styled-components": "^6.0.2",
    "@types/jest": "^29.5.14",
    "@types/styled-components": "^5.1.34",
    "babel-jest": "^29.0.2",
    "babel-plugin-styled-components": "^2.0.7",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.4.1",
    "jest-fail-on-console": "^3.3.1",
    "jest-styled-components": "^7.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "styled-components": "^5.3.11",
    "typescript": "^5.6.3"
  }
}
