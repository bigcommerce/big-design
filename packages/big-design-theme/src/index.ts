import { createHelpers, Helpers } from './helpers';
import { ThemeOptions, themeOptions } from './options';
import { Border, BorderRadius, createBorder, createBorderRadius } from './system/border';
import { Breakpoints, breakpoints, BreakpointValues, breakpointValues } from './system/breakpoints';
import { Colors, colors } from './system/colors';
import * as keyframes from './system/keyframes';
import { createLineHeight, LineHeight } from './system/line-height';
import { Shadow, shadow } from './system/shadow';
import { createSpacing, Spacing } from './system/spacing';
import { createTypography, Typography } from './system/typography';
import { ZIndex, zIndex } from './system/z-index';

export * from './helpers';
export * from './system';

export interface ThemeInterface {
  border: Border;
  borderRadius: BorderRadius;
  breakpointValues: BreakpointValues;
  breakpoints: Breakpoints;
  colors: Colors;
  helpers: Helpers;
  keyframes: typeof keyframes;
  lineHeight: LineHeight;
  shadow: Shadow;
  spacing: Spacing;
  typography: Typography;
  zIndex: ZIndex;
}

export const createTheme = (customOptions: Partial<ThemeOptions> = {}): ThemeInterface => {
  themeOptions.setOptions(customOptions);

  return {
    border: createBorder(),
    borderRadius: createBorderRadius(),
    breakpointValues,
    breakpoints,
    colors,
    helpers: createHelpers(),
    keyframes,
    lineHeight: createLineHeight(),
    shadow,
    spacing: createSpacing(),
    typography: createTypography(),
    zIndex,
  };
};

export const theme: ThemeInterface = createTheme();
