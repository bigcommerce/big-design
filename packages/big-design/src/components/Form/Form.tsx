import React, { ComponentPropsWithoutRef, forwardRef, Ref } from 'react';

import { MarginProps } from '../../helpers';
import { typedMemo } from '../../utils';

import { StyledForm } from './styled';
import { FormContext } from './useFormContext';

interface PrivateProps {
  readonly forwardedRef: Ref<HTMLFormElement>;
}

export type FormProps = ComponentPropsWithoutRef<'form'> &
  MarginProps & {
    readonly fullWidth?: boolean;
  };

const StyleableForm: React.FC<PrivateProps & FormProps> = ({ forwardedRef, ...props }) => {
  return (
    <FormContext.Provider value={{ fullWidth: props.fullWidth }}>
      <StyledForm {...props} ref={forwardedRef} />
    </FormContext.Provider>
  );
};

export const Form = typedMemo(
  forwardRef<HTMLFormElement, FormProps>(({ className, style, ...props }, ref) => (
    <StyleableForm {...props} forwardedRef={ref} />
  )),
);
