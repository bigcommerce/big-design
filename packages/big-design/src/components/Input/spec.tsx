import { AddIcon } from '@bigcommerce/big-design-icons';
import { theme } from '@bigcommerce/big-design-theme';
import { render, screen } from '@testing-library/react';
import React, { createRef } from 'react';
import 'jest-styled-components';

import { warning } from '../../utils';
import { FormControlDescription, FormControlError, FormControlLabel, FormGroup } from '../Form';

import { Input } from './index';

const basicBorderStyle = `1px solid ${theme.colors.secondary30}`;
const errorBorderStyle = `1px solid ${theme.colors.danger40}`;

test('forwards ref', () => {
  const ref = createRef<HTMLInputElement>();
  const { container } = render(<Input ref={ref} />);
  const input = container.querySelector('input');

  expect(ref.current).toBe(input);
});

test('renders an input tag', () => {
  const { container } = render(<Input />);

  expect(container.querySelector('input')).toBeInTheDocument();
});

test('renders an input with matched label', () => {
  render(<Input label="Test Label" />);

  // This one checks for matching id and htmlFor
  expect(screen.getByLabelText('Test Label')).toBeInTheDocument();
});

test('create unique ids if not provided', async () => {
  render(
    <>
      <Input data-testid="item1" label="Test Label" />
      <Input data-testid="item2" label="Test Label" />
    </>,
  );

  const item1 = await screen.findByTestId('item1');
  const item2 = await screen.findByTestId('item2');

  expect(item1).toBeInTheDocument();
  expect(item2).toBeInTheDocument();
  expect(item1.id).not.toBe(item2.id);
});

test('respects provided id', async () => {
  render(<Input data-testid="test-input" id="test" label="Test Label" />);

  const input = await screen.findByTestId<HTMLInputElement>('test-input');

  expect(input.id).toBe('test');
});

test('matches label htmlFor with id provided', async () => {
  render(<Input id="test" label="Test Label" />);

  const label = await screen.findByText<HTMLLabelElement>('Test Label');

  expect(label.htmlFor).toBe('test');
});

test('respects provided labelId', async () => {
  render(<Input label="Test Label" labelId="test" />);

  const label = await screen.findByText<HTMLLabelElement>('Test Label');

  expect(label.id).toBe('test');
});

test('renders a description', () => {
  const descriptionText = 'This is a description';
  render(<Input description={descriptionText} />);

  expect(screen.getByText(descriptionText)).toBeInTheDocument();
});

test('renders an error', () => {
  const errorText = 'This is an error';
  const { container } = render(
    <FormGroup>
      <Input error={errorText} />
    </FormGroup>,
  );
  const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

  expect(screen.getByText(errorText)).toBeInTheDocument();
  expect(styledInputWrapper).toHaveStyleRule('border', errorBorderStyle);
});

test('accepts a Label Component', () => {
  const CustomLabel = (
    <FormControlLabel>
      This is a custom Label
      <a data-testid="test" href="#">
        has a url
      </a>
    </FormControlLabel>
  );

  render(<Input label={CustomLabel} />);

  expect(screen.getByTestId('test')).toBeInTheDocument();
});

test('does not accept non-Label Components', () => {
  const NotALabel = (
    <div>
      This is a not custom Label Component
      <a data-testid="test" href="#">
        has a url
      </a>
    </div>
  );

  render(<Input label={NotALabel} />);

  expect(screen.queryByTestId('test')).not.toBeInTheDocument();
});

test('accepts a Description Component', () => {
  const CustomDescription = (
    <FormControlDescription>
      This is a custom Description
      <a data-testid="test" href="#">
        has a url
      </a>
    </FormControlDescription>
  );

  render(<Input description={CustomDescription} />);

  expect(screen.getByTestId('test')).toBeInTheDocument();
});

test('does not accept non-Description Components', () => {
  const NotADescription = (
    <div>
      This is a not custom description
      <a data-testid="test" href="#">
        has a url
      </a>
    </div>
  );

  render(<Input description={NotADescription} />);

  expect(screen.queryByTestId('test')).not.toBeInTheDocument();
});

test('accepts an Error Component', () => {
  const CustomError = (
    <FormControlError>
      This is a custom Error Component
      <a data-testid="test" href="#">
        has a url
      </a>
    </FormControlError>
  );

  const { container } = render(
    <FormGroup>
      <Input error={CustomError} />
    </FormGroup>,
  );
  const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

  expect(screen.getByTestId('test')).toBeInTheDocument();
  expect(styledInputWrapper).toHaveStyleRule('border', errorBorderStyle);
});

test('does not accept non-Error Components', () => {
  const NotAnError = (
    <div>
      This is a not a custom error component
      <a data-testid="test" href="#">
        has a url
      </a>
    </div>
  );

  const { container } = render(
    <FormGroup>
      <Input error={NotAnError} />
    </FormGroup>,
  );
  const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

  expect(screen.queryByTestId('test')).not.toBeInTheDocument();
  expect(styledInputWrapper).toHaveStyleRule('border', basicBorderStyle);
});

test('renders iconLeft', () => {
  render(<Input iconLeft={<AddIcon data-testid="icon" />} />);

  expect(screen.getByTestId('icon')).toBeInTheDocument();
});

test('renders iconRight', () => {
  render(<Input iconRight={<AddIcon data-testid="icon" />} />);

  expect(screen.getByTestId('icon')).toBeInTheDocument();
});

test('renders both icons', () => {
  render(
    <Input
      iconLeft={<AddIcon data-testid="icon-left" />}
      iconRight={<AddIcon data-testid="icon-right" />}
    />,
  );

  expect(screen.getByTestId('icon-left')).toBeInTheDocument();
  expect(screen.getByTestId('icon-right')).toBeInTheDocument();
});

test('renders all together', () => {
  const { container } = render(
    <Input
      description="This is a description"
      iconLeft={<AddIcon data-testid="icon-left" />}
      iconRight={<AddIcon data-testid="icon-right" />}
      label="This is a label"
    />,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('error shows with valid string', () => {
  const error = 'Error';
  const { container, rerender } = render(
    <FormGroup>
      <Input error="" />
    </FormGroup>,
  );
  const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

  expect(container.querySelector('[class*="StyledError"]')).not.toBeInTheDocument();
  expect(styledInputWrapper).toHaveStyleRule('border', basicBorderStyle);

  rerender(
    <FormGroup>
      <Input error={error} />
    </FormGroup>,
  );

  expect(container.querySelector('[class*="StyledError"]')).toBeInTheDocument();
  expect(styledInputWrapper).toHaveStyleRule('border', errorBorderStyle);
});

describe('error shows when an array of strings', () => {
  test('empty array', () => {
    const { container } = render(
      <FormGroup>
        <Input error={[]} />
      </FormGroup>,
    );
    const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

    expect(container.querySelector('[class*="StyledError"]')).not.toBeInTheDocument();
    expect(styledInputWrapper).toHaveStyleRule('border', basicBorderStyle);
  });

  test('array with empty strings', () => {
    const { container } = render(
      <FormGroup>
        <Input error={['', '']} />
      </FormGroup>,
    );
    const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

    expect(container.querySelector('[class*="StyledError"]')).not.toBeInTheDocument();
    expect(styledInputWrapper).toHaveStyleRule('border', basicBorderStyle);
  });

  test('array with valid strings', () => {
    const errors = ['Error 0', 'Error 1'];
    const { container, getByText } = render(
      <FormGroup>
        <Input error={errors} />
      </FormGroup>,
    );
    const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

    expect(styledInputWrapper).toHaveStyleRule('border', errorBorderStyle);

    errors.forEach((error) => expect(screen.getByText(error)).toBeInTheDocument());
  });
});

test('error shows when an array of Errors', () => {
  const testIds = ['error_0', 'error_1'];
  const errors = testIds.map((id) => (
    <FormControlError data-testid={id} key={id}>
      Error
    </FormControlError>
  ));
  const { container } = render(
    <FormGroup>
      <Input error={errors} />
    </FormGroup>,
  );
  const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

  testIds.forEach((id) => expect(screen.getByTestId(id)).toBeInTheDocument());

  expect(styledInputWrapper).toHaveStyleRule('border', errorBorderStyle);
});

describe('error does not show when invalid type', () => {
  test('single element', () => {
    const error = <div data-testid="err">Error</div>;
    const { container } = render(
      <FormGroup>
        <Input error={error} />
      </FormGroup>,
    );
    const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

    expect(warning).toHaveBeenCalledTimes(1);
    expect(screen.queryByTestId('err')).not.toBeInTheDocument();
    expect(styledInputWrapper).toHaveStyleRule('border', basicBorderStyle);
  });

  test('array of elements', () => {
    const errors = [
      'Error',
      <FormControlError key="1">Error</FormControlError>,
      <div data-testid="err" key="2">
        Error
      </div>,
    ];

    const { container } = render(
      <FormGroup>
        <Input error={errors} />
      </FormGroup>,
    );
    const styledInputWrapper = container.querySelector('[class*="StyledInputWrapper"]');

    expect(warning).toHaveBeenCalledTimes(1);
    expect(screen.queryByTestId('err')).not.toBeInTheDocument();
    expect(styledInputWrapper).toHaveStyleRule('border', errorBorderStyle);
  });
});

test('appends * text to label if input is required', () => {
  render(<Input label="label" required />);

  expect(screen.getByLabelText('label *')).toBeInTheDocument();
});
