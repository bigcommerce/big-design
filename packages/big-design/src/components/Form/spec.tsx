import { fireEvent, render, screen } from '@testing-library/react';
import React, { createRef } from 'react';
import 'jest-styled-components';

import { Fieldset } from '../Fieldset';
import { Input } from '../Input';

import { Form, FormGroup } from './index';

test('forwards ref', () => {
  const ref = createRef<HTMLFormElement>();

  render(<Form aria-label="Test form" ref={ref} />);

  const form = screen.getByRole('form');

  expect(form).toBe(ref.current);
});

test('calls onSubmit', async () => {
  const onSubmit = jest.fn();

  render(<Form aria-label="Test form" name="testForm" onSubmit={onSubmit} />);

  const form = await screen.findByRole('form');

  fireEvent.submit(form);

  expect(onSubmit).toHaveBeenCalled();
});

test('simple form render', () => {
  render(
    <Form aria-label="Test form">
      <Fieldset
        description="Minim velit quis aute adipisicing adipisicing do do exercitation cupidatat enim ex voluptate consequat labore."
        legend="Primary contact"
      >
        <FormGroup>
          <Input
            description="This is an example description for First Name"
            label="First Name"
            placeholder="Placeholder text"
          />
        </FormGroup>

        <FormGroup>
          <Input
            description="This is an example description for Last Name. Featuring a Left Icon."
            label="Middle Name"
            placeholder="Placeholder text"
          />
        </FormGroup>
      </Fieldset>
    </Form>,
  );

  const form = screen.getByRole('form');

  expect(form).toMatchSnapshot();
});

test('has margin props', () => {
  const { rerender } = render(<Form aria-label="Test form" />);

  const form = screen.getByRole('form');

  expect(form).not.toHaveStyle('margin: 1rem');

  rerender(<Form aria-label="Test form" margin="medium" />);

  expect(form).toHaveStyle('margin: 1rem');
});
