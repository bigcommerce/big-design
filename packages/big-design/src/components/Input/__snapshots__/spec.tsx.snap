// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders all together 1`] = `
.c5 {
  vertical-align: middle;
  height: 1.5rem;
  width: 1.5rem;
}

.c0 {
  color: #313440;
  margin: 0 0 1rem;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5rem;
  margin: 0 0 0.5rem;
}

.c2 {
  color: #313440;
  margin: 0 0 1rem;
  color: #5E637A;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.25rem;
  margin: 0 0 0.75rem;
}

.c2:last-child {
  margin-bottom: 0;
}

.c3 {
  -webkit-transition: all 150ms ease-out;
  transition: all 150ms ease-out;
  -webkit-transition-property: border,box-shadow;
  transition-property: border,box-shadow;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFFFFF;
  border-radius: 0.25rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 2.25rem;
  overflow: hidden;
  position: relative;
  width: 100%;
  border: 1px solid #D9DCE9;
}

.c3:hover:not([disabled]) {
  border: 1px solid #B4BAD1;
}

.c3[disabled] {
  background-color: #ECEEF5;
}

.c7 {
  background-color: inherit;
  border: 0;
  box-sizing: border-box;
  color: #313440;
  -webkit-flex: 1 1 40%;
  -ms-flex: 1 1 40%;
  flex: 1 1 40%;
  height: 100%;
  padding: 0;
  padding-left: 0.5rem;
  width: 100%;
  padding-right: 2.25rem;
  padding-left: 2.25rem;
  min-height: 2.125rem;
}

.c7:focus {
  outline: none;
}

.c7::-webkit-input-placeholder {
  color: #B4BAD1;
  font-size: 1rem;
}

.c7::-moz-placeholder {
  color: #B4BAD1;
  font-size: 1rem;
}

.c7:-ms-input-placeholder {
  color: #B4BAD1;
  font-size: 1rem;
}

.c7::placeholder {
  color: #B4BAD1;
  font-size: 1rem;
}

.c7:-webkit-autofill {
  background-color: #F0F3FF !important;
  -webkit-box-shadow: 0 0 0px 1000px #F0F3FF inset;
}

.c7[disabled] {
  background-color: #ECEEF5;
  color: #5E637A;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #5E637A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  position: absolute;
  top: 0;
  padding-right: 0.25rem;
  padding-left: 0.5rem;
  left: 0;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #5E637A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  position: absolute;
  top: 0;
  padding-right: 0.5rem;
  padding-left: 0.25rem;
  right: 0;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 100%;
}

.c1 {
  cursor: pointer;
  display: inline-block;
  margin-bottom: 0.25rem;
}

<div>
  <label
    class="c0 c1"
    for=":rp:"
  >
    This is a label
  </label>
  <p
    class="c2"
  >
    This is a description
  </p>
  <span
    class="c3"
  >
    <div
      class="c4"
    >
      <svg
        aria-hidden="true"
        class="c5"
        data-testid="icon-left"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1"
        />
      </svg>
    </div>
    <div
      class="c6"
    >
      <input
        class="c7"
        id=":rp:"
      />
    </div>
    <div
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c5"
        data-testid="icon-right"
        fill="currentColor"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1"
        />
      </svg>
    </div>
  </span>
</div>
`;
